src/cascadia/QueryExtension/GithubCopilotLLMProvider.cpp

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "GithubCopilotLLMProvider.h"
+#include "../../types/inc/utils.hpp"
+#include "LibraryResources.h"
+#include "WindowsTerminalIDAndSecret.h"
+
+#include "GithubCopilotLLMProvider.g.cpp"
+
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::UI::Core;
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Xaml::Controls;
+using namespace winrt::Windows::System;
+namespace WWH = ::winrt::Windows::Web::Http;
+namespace WSS = ::winrt::Windows::Storage::Streams;
+namespace WDJ = ::winrt::Windows::Data::Json;
+
+// branding data
+static constexpr wil::zwstring_view headerIconPath{ L"ms-appx:///ProfileIcons/githubCopilotLogo.png" };
+static constexpr wil::zwstring_view badgeIconPath{ L"ms-appx:///ProfileIcons/githubCopilotBadge.png" };
+
+// header and request strings
+static constexpr std::wstring_view applicationJsonString{ L"application/json" };
+static constexpr std::wstring_view bearerString{ L"Bearer" };
+static constexpr std::wstring_view copilotIntegrationIdString{ L"Copilot-Integration-Id" };
+static constexpr std::wstring_view clientIdKey{ L"client_id" };
+static constexpr std::wstring_view clientSecretKey{ L"client_secret" };
+static constexpr std::wstring_view endpointAndUsernameRequestString{ L"{ viewer { copilotEndpoints { api } login } }" };
+
+// json keys
+static constexpr std::wstring_view accessTokenKey{ L"access_token" };
+static constexpr std::wstring_view refreshTokenKey{ L"refresh_token" };
+static constexpr std::wstring_view stateKey{ L"state" };
+static constexpr std::wstring_view urlKey{ L"url" };
+static constexpr std::wstring_view queryKey{ L"query" };
+static constexpr std::wstring_view codeKey{ L"code" };
+static constexpr std::wstring_view errorKey{ L"error" };
+static constexpr std::wstring_view errorDescriptionKey{ L"error_description" };
+static constexpr std::wstring_view dataKey{ L"data" };
+static constexpr std::wstring_view apiKey{ L"api" };
+static constexpr std::wstring_view viewerKey{ L"viewer" };
+static constexpr std::wstring_view copilotEndpointsKey{ L"copilotEndpoints" };
+static constexpr std::wstring_view loginKey{ L"login" };
+static constexpr std::wstring_view grantTypeKey{ L"grant_type" };
+static constexpr std::wstring_view contentKey{ L"content" };
+static constexpr std::wstring_view messageKey{ L"message" };
+static constexpr std::wstring_view messagesKey{ L"messages" };
+static constexpr std::wstring_view choicesKey{ L"choices" };
+static constexpr std::wstring_view roleKey{ L"role" };
+static constexpr std::wstring_view assistantKey{ L"assistant" };
+static constexpr std::wstring_view userKey{ L"user" };
+static constexpr std::wstring_view systemKey{ L"system" };
+
+// endpoints
+static constexpr std::wstring_view githubGraphQLEndpoint{ L"https://api.github.com/graphql" };
```

Comment:
this will not be different for enterprise redirection?

Reply:
Enterprise redirection is just for the copilot endpoint itself (i.e. the LLM that answers questions), this URL is just to obtain the user's username and shouldn't be affected